include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -2565,8 +2566,11 @@ inline unsigned int
 DoFAccessor<0, DoFHandlerType<1, spacedim>, level_dof_access>::
   n_active_fe_indices() const
 {
-  Assert(false, ExcNotImplemented());
-  return numbers::invalid_unsigned_int;
+  const bool check = std::is_same<DoFHandlerType<1, spacedim>,
+                                  dealii::DoFHandler<1, spacedim>>::value;
+  Assert(check == true, ExcNotImplemented());
```

Comment:
Can you move the `std::is_same...` into the assert? This avoids introducing the variable `check` that is unused in release mode.

Reply:
No, I cannot. I have tried it. But the compiler complained:
```bash
dof_accessor.templates.h:2570:103: error: macro "Assert" passed 5 arguments, but takes just 2
                                   dealii::DoFHandler<1, spacedim>>::value == true, ExcNotImplemented());
                                                                                                       ^
```
Do you know a workaround?

Reply:
Try to put it in braces, `(std::is_same<...>)`.

Reply:
There is also the following idiom:
```
#ifdef DEBUG
    // ...
    Assert(...);
#endif
```

Reply:
@kronbichler: That worked! Stupid that I did not think of that myself...